alerts/honeycomb.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# Copyright (c) 2014 Mozilla Corporation
+
+from lib.alerttask import AlertTask
+from query_models import SearchQuery, TermMatch
+import re
+
+
+class AlertHoneycomb(AlertTask):
+    def main(self):
+        # look for events in last 10 mins
+        search_query = SearchQuery(minutes=10)
+
+        search_query.add_must([
+            TermMatch('category', 'syslog'),
+            TermMatch('processname', 'Honeycomb')
+        ])
+
+        self.filtersManual(search_query)
+
+        # Search aggregations on field 'hostname', keep X samples of
+        # events at most
+        self.searchEventsAggregated('hostname', samplesLimit=10)
+        # alert when >= X matching events in an aggregation
+        # I think it makes sense to alert every time here
+        self.walkAggregations(threshold=1)
+
+    # Set alert properties
+    def onAggregation(self, aggreg):
+        # aggreg['count']: number of items in the aggregation, ex: number of failed login attempts
+        # aggreg['value']: value of the aggregation field, ex: toto@example.com
+        # aggreg['events']: list of events in the aggregation
+        category = 'honeypot'
+        tags = ['honeypot', 'honeycomb']
+        severity = 'WARNING'
+
+        # This is required in order to extract and display the source IP accessing the honeypot
+        pattern = '/originating_ip="(\d+\.\d+\.\d+\.\d+)"/'
+        offendingIPs = []
+
+        for event in aggreg['allevents']:
+            offendingIPs.append(re.search(pattern, aggreg['allevents'][event]['_source']['summary']))
```

Comment:
I wonder if we need some more logic here to cover a case where the data within summary is not what we expect.

Say summary contained "foo", would this stack trace mozdef or handle it gracefully?

Reply:
I think you mean, if the event summary is completely different than what expect, i.e. no originatingIP, but simply a random string like "foo". Still I don't think it would stack trace (not from regex at least), but we should probably but a validation the next line in case there is no match (i.e. len(offendingIP) == 0)

Reply:
I have included a condition from L48.

Reply:
This is actually the condition I was thinking about (though not a stack trace, certainly non-desirable)...

>>> array = [1,2,3]
>>> array.append(re.search('foo', 'bar'))
>>> array
[1, 2, 3, None]

Reply:
Added another condition in https://github.com/mozilla/MozDef/pull/747/commits/81738efeffecad9d288589a63905f37e4285e87e